Automattic/simplenote-ios
# Tags List Refresh: Mark III

### Details:
In this PR we're enhancing the Tags List:

- New **Untagged Notes** filter
- The Tags section will no longer be displayed, whenever there are no tags
- Encapsulated all of the filtering NSPredicate(s)
- And we're getting new Unit Tests!

@danielebogo Daniele!! May I bug you with this one?
**Thank you!!!**

Ref. #408

### Testing: Hiding Tags Sections
1. Log into a testing account with no Tag

- [x] Verify that an account with no Tags doesn't show the `Tags` section
- [x] Verify that an account with no Tags doesn't show the `Untagged` section

### Testing: Predicates!
1. Log into your account
2. Add three notes (with random contents)
3. Tag one of the new notes
4. Trash one of the new notes
5. And of course, leave one of them untagged

- [x] Verify the `Trash` filter shows the trashed note
- [x] Verify the `Untagged` filter shows the untagged note
- [x] Verify that tapping over the tag you've set in (3) properly displays the new note
- [x] Verify the new unit tests look green!


---

### Screenshots:

<img src="https://user-images.githubusercontent.com/1195260/66854243-5e03a080-ef57-11e9-87fc-ae27b15bfe0c.png" width="200">
